### PR TITLE
Polling 3.0

### PR DESCRIPTION
**Pull request by marianozicavo**
Polling!

This basically improves `completeState`, `completeReducer` and `completeTypes` to support polling.

You can look at #44 for an extended explanation.
Also look at the diagram and the recipes documentation.

### CODE REVIEW

File: recipes/POLLING.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# Basic polling recipe
+
+Imagine you have to translate a PDF. In order to translate it, you will upload it to a server in order for it translate it and create a new URL where you can fetch the new translated PDF.
```
Comment (guidoprinc) [inline recipes/POLLING.md:3]: `... in order for it to translate ...`

File: recipes/POLLING.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+If at any time the server responds with a `200`, `currentTranslationStatus` will hold the server response and `currentTranslationStatusIsRetrying` will be set to `false`. This marks the polling as being over.
+
+If at any time the server responds with any other HTTP status, `currentTranslationStatusError` will hold the server response and `currentTranslationStatusIsRetrying` will be set to `false`. This also marks the polling as being over.
+
+Here's a diagram to help you further understand how the polling works
+![alt text](./polling.png "Logo Title Text 1")
+
+## Basic polling recipe with retries limit
+
+You can add an easy linit to the number of retries with:
```
Comment (guidoprinc) [inline recipes/POLLING.md:72]: limit

File: src/completers/completeTypes/test.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import completeTypes from '.';
 
 describe('completeTypes', () => {
-  it('Completes from an array\'s element', () => {
+  it("Completes from an array's element", () => {
```
Comment (guidoprinc) [inline src/completers/completeTypes/test.js:4]: why double quotes?

File: src/effects/onCancel/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+## onCancel - Effect
+
+This effect is cancels the polling flow.
```
Comment (guidoprinc) [inline src/effects/onCancel/README.md:3]: remove the `is`

**Review by guidoprinc — CHANGES_REQUESTED** [review]

File: src/completers/completeTypes/test.js (hunk 3 above)
Comment (marianozicavo) [reply src/completers/completeTypes/test.js:4]: So I don't have to use \' and it's easier to read, I think

**Review by guidoprinc — APPROVED** [review]

**Pull request merged**